cmd/envoy/main.go

```diff
@@ -574,46 +574,53 @@ func daybreakImport(c *cli.Context) (err error) {
 			log.Warn().Msg(fmt.Sprintf("Protocol must be 'sunrise' to import: '%s' (DirID: '%s')", apiCounterparty.Name, apiCounterparty.DirectoryID))
 			continue
 		}
+		if modelCounterparty.RegisteredDirectory.Valid && modelCounterparty.RegisteredDirectory.String != "daybreak.rotational.io" {
+			log.Warn().Msg(fmt.Sprintf("RegisteredDirectory must be 'daybreak.rotational.io' to import: '%s' (DirID: '%s')", apiCounterparty.Name, apiCounterparty.DirectoryID))
+			continue
+		}
 
 		if modelCounterparty.DirectoryID.Valid {
 			cSrc, ok := srcMap[modelCounterparty.DirectoryID.String]
 			if ok {
 				// counterparty is present in DB, so we update it
 				modelCounterparty.ID = cSrc.ID
-				if err = db.UpdateCounterparty(ctx, modelCounterparty); err != nil {
+				apiCounterparty.ID = cSrc.ID
+				if err = UpdateCounterpartyAndContacts(ctx, modelCounterparty, apiCounterparty); err != nil {
 					log.Warn().Err(err).Msg(fmt.Sprintf("Error when updating Counterparty: '%s' (ID: '%s')", modelCounterparty.Name, modelCounterparty.ID))
```

Comment:
I would recommend this code:

```suggestion
					log.Warn().Err(err).Str("directory_id", apiCounterparty.DirectoryID).Str("name", apiCounterparty.Name).Msg("could not convert counterparty to model")
```

When we run this in production, the JSON format will be a bit easier to parse and handle in Google Log Explorer in the console. The human readable version isn't too bad either, it's very similar to the string format you have; without the parentheses (and maybe equal instead of colon)